src/traversals/iterators.jl

```diff
@@ -0,0 +1,130 @@
+"""
+    abstract type IteratorAlgorithm
+
+`IteratorAlgorithm` is an abstract type which specifies a particular algorithm to use when iterating through a graph.
+"""
+abstract type IteratorAlgorithm end
```

Comment:
```suggestion
abstract type VertexIterator end
```
I think this makes it more clear, that we iterate over vertices.

Reply:
Agreed, I'll update and perhaps wait on thoughts on my reply above before making another commit to possible to include that as well. Like I said in the above reply, I'm not too well verse in graphs so this helps a lot. Thanks

Reply:
Also, we might parametrize the Iterator in the type of vertex that we return:
```julia
abstract type VertexIterator{T <: Integer} end
```

Reply:
Sure. I didn't realize there are other types of vertices.. can you point me to an example so I can read up